packages/mdc-base/component.ts

```diff
@@ -82,16 +85,16 @@ class MDCComponent<FoundationType extends MDCFoundation> {
    * Wrapper method to add an event listener to the component's root element. This is most useful when
    * listening for custom events.
    */
-  listen(evtType: string, handler: EventListener) {
-    this.root_.addEventListener(evtType, handler);
+  listen<T extends Event>(evtType: string, handler: EventListener | CustomEventListener<T>) {
```

Comment:
We recently [added some types](https://github.com/material-components/material-components-web/blob/feat/typescript/packages/mdc-dom/events.ts) for dealing with standard browser events, if you also want to give users the ability to specify the event type in the case of custom events, you could use typescript's function overloading to support it while still filling in the right type for standard events [(example)](http://www.typescriptlang.org/play/#src=type%20EventType%20%3D%20keyof%20GlobalEventHandlersEventMap%3B%0D%0Atype%20SpecificEventListener%3CK%20extends%20EventType%3E%20%3D%20(evt%3A%20GlobalEventHandlersEventMap%5BK%5D)%20%3D%3E%20void%3B%0D%0Atype%20CustomEventListener%3CE%20extends%20Event%3E%20%3D%20(evt%3A%20E)%20%3D%3E%20void%3B%0D%0A%0D%0Afunction%20listen%3CK%20extends%20EventType%3E(event%3A%20K%2C%20listener%3A%20SpecificEventListener%3CK%3E)%3B%0D%0Afunction%20listen%3CE%20extends%20Event%3E(event%3A%20string%2C%20listener%3A%20CustomEventListener%3CE%3E)%3B%0D%0Afunction%20listen(event%2C%20listener)%20%7B%20%2F*%20implementation%20here%20*%2F%20%7D%0D%0A%0D%0Ainterface%20WhateverEvent%20extends%20Event%20%7B%0D%0A%20%20%20%20stuff%3A%20string%3B%0D%0A%7D%0D%0A%0D%0Alisten('click'%2C%20e%20%3D%3E%20e.clientX)%3B%0D%0Alisten%3CWhateverEvent%3E('whatever'%2C%20e%20%3D%3E%20e.stuff)%3B)

Reply:
Great suggestion, @mmalerba! PTAL 😀